opencensus/exporters/trace/zipkin/zipkin_exporter.cc

```diff
@@ -0,0 +1,342 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "zipkin_exporter.h"
+
+#include <curl/curl.h>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+#include "absl/strings/str_cat.h"
+
+namespace opencensus {
+namespace exporters {
+namespace trace {
+
+namespace {
+
+constexpr const char *kZipkinLibname = "zipkin";
+constexpr const char *kZkipkinVersion = "2.0";
+
+void SerializeJson(rapidjson::Writer<rapidjson::StringBuffer> *writer,
+                   const ::opencensus::trace::exporter::SpanData &span) {
+  writer->StartObject();
+
+  writer->Key("name");
+  writer->String(std::string(span.name()));
+
+  writer->Key("traceId");
+  writer->String(span.context().trace_id().ToHex());
+
+  writer->Key("id");
+  writer->String(span.context().span_id().ToHex());
+
+  writer->Key("parentId");
+  writer->String(span.parent_span_id().ToHex());
+
+  writer->Key("annotations");
+  writer->StartArray();
+  for (const auto &annotation : span.annotations().events()) {
+    writer->StartObject();
+    writer->Key("timestamp");
+    writer->Int64(absl::ToUnixMicros(annotation.timestamp()));
+    writer->Key("value");
+    writer->String(annotation.event().DebugString());
+    writer->EndObject();
+  }
+  writer->EndArray(span.annotations().events().size());
+
+  writer->Key("binaryAnnotations");
```

Comment:
let's not do v1 format, or somehow permit v2 format. binaryAnnotations is archaic and replaced by "tags" which are always string -> string map  https://zipkin.io/zipkin-api/#/default/post_spans

Reply:
Done.